Automattic/sensei
# Add locked lesson states

Part of #4445

### Changes proposed in this Pull Request

* It introduces icon support for the Context Notices (Maybe the [approach can change in the future](https://github.com/Automattic/sensei/issues/4482)).
* It also adds new notices for lessons with prerequisites, and lessons with prerequisites awaiting grade.

### Testing instructions


* Enable the feature flag: `add_filter( 'sensei_feature_flag_course_theme', '__return_true' );`
* Create a course.
* Add 2 pairs of lessons and lesson prerequisites (you can set the prerequisites in the lesson sidebar):
  * In the first one, the prerequisite should contain a quiz with pass required and without auto-grade.
  * In the second one, just a simple lesson.
* Select the "Sensei Theme" as the "Course Theme" (sidebar).
* As a student, complete the quiz, but don't grade it.
* Make sure the lesson that contains a simple lesson as a prerequisite shows a message of lesson prerequisite (see screenshot with the message).
* Make sure the lesson that contains a quiz as a prerequisite shows an ungraded quiz message (see screenshot with the message).


### Screenshot / Video

<img width="784" alt="Screen Shot 2021-12-06 at 17 05 05" src="https://user-images.githubusercontent.com/876340/144914959-a1bd5f42-6cfd-447c-bf06-b11946c28077.png">

<img width="836" alt="Screen Shot 2021-12-07 at 10 56 15" src="https://user-images.githubusercontent.com/876340/145042084-0718b207-3e24-42c0-a482-365bb0abe922.png">

This message is a little different than the version from the design. New suggestions are welcome here. Context: p1638814484006200-slack-C02KZBEJ30D cc @sruj09
